### ISSUE TITLE
Dynamic disk space check for anago

### ISSUE THREAD

**Issue by david-mcmahon** [kubernetes/release#114]
Right now the anago release tool has a disk check in it to make sure we have enough disk to do a full release build, but it's static and the state of Kubernetes (and how big it is) is wildly in flux 

I'd like to dynamically set the disk requirements based on a recent continuous build directory (cross-build?).  

Looking for a number like this:

```
# From a recent release build on my machine
$ df -sh anago-v1.5.0-alpha.1
32G anago-v1.5.0-alpha.1
```
## 

Some thoughts from the seeding email thread:

@ixdy said: 

The crossbuild is happening inside docker, and there are likely simultaneous builds happening at the same time. We might be able to query docker for what the volume sizes are, though I'm not sure exactly how best to do that.
## 

As some point of reference for places we might look for info, cross-build is currently running on agent-heavy-6.

```
# docker ps -a -s
CONTAINER ID        IMAGE                         COMMAND                  CREATED             STATUS                         PORTS               NAMES                        SIZE
d3e8badf2d4d        kube-build:build-6f4b9d8412   "make cross"             19 minutes ago      Up 18 minutes                                      kube-build-6f4b9d8412        268.9 MB (virtual 2.292 GB)
262c23accbb7        kube-build:build-6f4b9d8412   "chown -R 1014.1014 /"   19 minutes ago      Exited (0) 19 minutes ago                          kube-build-data-6f4b9d8412   0 B (virtual 2.049 GB)
b92f1e75aa75        kube-build:build-e1f19b0056   "chown -R 1014.1014 /"   About an hour ago   Exited (0) About an hour ago                       kube-build-data-e1f19b0056   0 B (virtual 2.049 GB)
5fc2028614c7        kube-build:build-eb31ccc440   "true"                   About an hour ago   Exited (0) About an hour ago                       kube-build-data-eb31ccc440   0 B (virtual 1.676 GB)
d4ac7913d125        kube-build:build-2c9f945771   "chown -R 0.0 /go/src"   2 hours ago         Exited (0) 2 hours ago                             kube-build-data-2c9f945771   0 B (virtual 1.944 GB)
2267a26d712f        kube-build:build-6bf291a109   "chown -R 0.0 /go/src"   2 hours ago         Exited (0) 2 hours ago                             kube-build-data-6bf291a109   0 B (virtual 1.944 GB)
root@agent-heavy-6:/var/lib/jenkins/workspace/kubernetes-cross-build# docker ps -a -s | grep 6f4b9d8412
d3e8badf2d4d        kube-build:build-6f4b9d8412   "make cross"             20 minutes ago      Up 19 minutes                                      kube-build-6f4b9d8412        498.2 MB (virtual 2.532 GB)
262c23accbb7        kube-build:build-6f4b9d8412   "chown -R 1014.1014 /"   20 minutes ago      Exited (0) 20 minutes ago                          kube-build-data-6f4b9d8412   0 B (virtual 2.049 GB)

# du -sh _output
3.3G    _output
```

We could maybe run `du` on the _output dir at the end of runs and save that in metadata somewhere? Not perfect, but it's a start of a metric.
## 

Disk usage shoots up a bit when producing the docker images and tarballs:

```
5.9G    _output/dockerized
20M     _output/images
9.2G    _output/release-stage
2.6G    _output/release-tars
```

Anyway, we should probably figure out some way of saving the size of _output in structured metadata. At the very least, we could just run `du -sh _output` in the build script, so a human can periodically easily update the thresholds. (We already run sha256sum for presumably similar reasons.)

cc @kubernetes/test-infra-maintainers

**Comment by mattymo** [comment]
Solving this problem in terms of comparing against local recent builds (which may or may have not been successful) is probably not the best approach. A safe value is $LAST_BUILD * 1.5. If your 32G size is representative of a complete build, then we should set the new disk check to 48GB (or round up to 50GB).

But I definitely agree with you @david-mcmahon that it is nice to have a structured metadata piece to capture how much build space was required at the end of buildtime that can be used by a human to update the threshold.

**Comment by david-mcmahon** [comment]
Getting a value from a *successful* recent build should be fine.  We could even average the last N + 10% or something.  The fine tuning is easy once the mechanism is in place.
